### PR TITLE
Added some seq operations

### PR DESCRIPTION
**Pull request by br4sco**
Added operations over sequences for appending, reversing, concatenation and foldr

### CODE REVIEW

File: stdlib/seq.mc [hunk 1]
```diff
@@ -31,6 +43,13 @@ let all = fix (lam all. lam p. lam xs.
   then true
   else and (p (nth xs 0)) (all p (slice xs 1 (length xs))))
 
+-- Append and concat
+
+let append = lam seq1. lam seq2.
+    let f = lam acc. lam x. cons x acc in
+    foldr f seq2 seq1
+
+let concat = lam seqs. foldl append [] seqs
```
Comment (EliasC) [inline stdlib/seq.mc]: `append` already exists in the built-ins with the name `concat`. I suggest replacing this chunk by:
```
let join = lam  seqs. foldl concat [] seqs
```

File: stdlib/seq.mc [hunk 2]
```diff
@@ -14,14 +14,26 @@ let foldl = fix (lam foldl. lam f. lam acc. lam seq.
...
 
+let foldr = fix (lam foldr. lam f. lam acc. lam seq.
+    if null seq
+    then acc
+    else f (foldr f acc (tail seq)) (head seq)
+)
+
+let rev = lam seq.
+    let f = lam acc. lam x. cons x acc in
+    foldl f [] seq
```
Comment (EliasC) [inline stdlib/seq.mc]: There is already a built-in version of this called `reverse`.

File: stdlib/seq.mc [hunk 3]
```diff
@@ -14,14 +14,26 @@ let foldl = fix (lam foldl. lam f. lam acc. lam seq.
...
+
+let foldr1 = lam f. lam seq. foldl1 f (reverse seq)
```
Comment (EliasC) [inline stdlib/seq.mc:25]: Not sure if `foldr f (head seq) (tail seq)` would be faster (no reverse, but also not tail-recursive).

File: stdlib/seq.mc [hunk 4]
```diff
@@ -14,14 +14,26 @@ let foldl = fix (lam foldl. lam f. lam acc. lam seq.
 )
 let foldl1 = lam f. lam l. foldl f (head l) (tail l)
 
+let foldr = fix (lam foldr. lam f. lam acc. lam seq.
+    if null seq
+    then acc
+    else f (foldr f acc (tail seq)) (head seq)
```
Comment (EliasC) [inline stdlib/seq.mc]: Should the arguments to `f` be flipped? My feeling is that `foldr cons [] [1,2,3]` should be `[1,2,3]`, whereas now it is an error.

**Comment by EliasC** [comment]
Resolved some conflicts, so you might want to pull before addressing any issues!

File: stdlib/seq.mc (hunk 3 above)
Comment (br4sco) [reply stdlib/seq.mc:25]: If you do `head [1,2,3]` you get `1` but when folding from right you want `3` to be the first value of the accumulator right?

File: stdlib/seq.mc (hunk 4 above)
Comment (br4sco) [reply stdlib/seq.mc]: I guess we could, but I think the above definition is consistent with foldl, i.e. `f : acc -> x -> acc` so `foldl cons [] [1,2,3]` also gives an error.

File: stdlib/seq.mc (hunk 3 above)
Comment (EliasC) [reply stdlib/seq.mc:25]: You're right! Maybe you have to implement it from scratch to get rid of the reverse. Let's just leave as is then!

File: stdlib/seq.mc (hunk 4 above)
Comment (br4sco) [reply stdlib/seq.mc]: No your right, for `foldr` we should have  `f: x -> acc -> acc` but for `foldl` then `f: acc -> x -> acc`.

**Comment by EliasC** [comment]
Looks good! Merging...

**Pull request merged**